tests/test_client.py

```diff
@@ -0,0 +1,184 @@
+import pytest
+import testinfra
+import subprocess
+import os
+import time
+import json
+
+
+@pytest.fixture
+def entities():
+    return {
+        'organization_label': 'Test_Organization',
+        'activation_key': 'Test AK',
+        'product': 'Test Product',
+        'product_label': 'Test_Product',
+        'yum_repository': 'Zoo',
+        'yum_repository_label': 'Zoo',
+        'errata_id': 'RHEA-2012:0055',
+        'container_repository_label': 'foremanbusybox'
+    }
+
+@pytest.fixture(scope='session')
+def client(request):
+    name = "smoker-client-%s" % request.param.replace('/', '_').replace(':', '_')
+    podman_id = subprocess.check_output(['podman', 'ps', '-q', '--filter', "name=%s" % name])
+
+    if not podman_id:
+        subprocess.check_output(['podman', 'run', '--name', name, '-dt', request.param]).decode().strip()
+
+    # return a testinfra connection to the container
+    yield testinfra.get_host("podman://" + name)
+
+@pytest.fixture(scope='session')
+def smoker():
+    return testinfra.get_host("local://localhost")
+
+@pytest.fixture
+def server_hostname(variables):
+    assert variables['server_hostname']
+    return variables['server_hostname']
+
+@pytest.fixture
+def foreman_version(variables):
+    assert variables['foreman_version']
+    return variables['foreman_version']
+
+def pytest_generate_tests(metafunc):
+    with open(metafunc.config.option.variables[0]) as file:
+      variables = json.load(file)
+      metafunc.parametrize('client', variables['clients'], indirect=True, scope='session')
```

Comment:
No need do this while the file is open
```suggestion
    metafunc.parametrize('client', variables['clients'], indirect=True, scope='session')
```